### PR TITLE
log err when remove container fails in rescheduling

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Sun Hongliang <allen.sun@daocloud.io>

### CODE REVIEW

File: cluster/watchdog.go [hunk 1]
```diff
@@ -47,7 +47,9 @@ func (w *Watchdog) removeDuplicateContainers(e *Engine) {
 			if containerInCluster.Config.SwarmID() == container.Config.SwarmID() && containerInCluster.Engine.ID != container.Engine.ID {
 				log.Debugf("container %s was rescheduled on node %s, removing it", container.Id, containerInCluster.Engine.ID)
 				// container already exists in the cluster, destroy it
-				e.RemoveContainer(container, true, true)
+				if err := e.RemoveContainer(container, true, true); err != nil {
+					log.Errorf("Failed to remove duplicate container %s on node %s: %v", container.Id, containerInCluster.Engine.ID, err)
```
Comment (dongluochen) [inline cluster/watchdog.go]: I think printing engine name improves readability. It's possible multiple engine has same name, but that's bad practice.

Comment (allencloud) [inline cluster/watchdog.go]: I changed two places in engine name.

**Comment by dongluochen** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Pull request merged**